gitlab/client.py

```diff
@@ -952,16 +953,26 @@ def should_emit_warning() -> bool:
         # Warn the user that they are only going to retrieve `per_page`
         # maximum items. This is a common cause of issues filed.
         total_items = "many" if gl_list.total is None else gl_list.total
-        utils.warn(
-            message=(
+        if cli:
+            message = (
+                f"Your query returned {len(items)} of {total_items}. To return all "
+                f"items use `--get-all`. To silence this warning use `--no-get-all`."
+            )
+            show_caller = False
```

Comment:
Is there a way we can just pass the message specifics (e.g. `message_details`) here, rather than a CLI boolean. This way we

1. Keep the API client and CLI decoupled
2. Allow passing other messages from non-`list()` calls like we had in a previous discussion. For example, 

We can keep a small default message with the pagination link with `list()`, but allow specifying a custom message from other callers.